src/Plugin/Plugins/TempHumidPlugin.cpp

```diff
@@ -0,0 +1,259 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Reads temperature and humidity from a sensor and displays it
+ * @author Flavio Curti <fcu-github@no-way.org>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "TempHumidPlugin.h"
+#include "FileSystem.h"
+
+#include <Board.h>
+#include <Color.h>
+#include <DHTesp.h>
+#include <Logging.h>
+
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+/* Initialize image path for temperature (scale) icon. */
+const char* TempHumidPlugin::IMAGE_PATH_TEMP_ICON      = "/images/temp.bmp";
+
+/* Initialize image path for humidity (drop) icon. */
+const char* TempHumidPlugin::IMAGE_PATH_HUMID_ICON     = "/images/drop.bmp";
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+void TempHumidPlugin::setSlot(const ISlotPlugin* slotInterf)
+{
+    m_slotInterf = slotInterf;
+    return;
+}
+
+void TempHumidPlugin::active(IGfx& gfx)
+{
+    lock();
+
+    // set time to show page - either 10s or slot_time / 4
+    // read here because otherwise we do not get config changes in slot_time
+    if (nullptr == m_slotInterf) {
+        m_pageTime = 10000U;
+    } else {
+        m_pageTime = m_slotInterf->getDuration() / 4;
```

Comment:
Add "U" to the 4 for type safety.

And why 4? Is this by intention or should it be 2, because of two available pages?